src/audio/module_adapter/module/volume/volume_generic.c

```diff
@@ -147,6 +188,46 @@ static void vol_s32_to_s32(struct processing_module *mod, struct input_stream_bu
 		y = audio_stream_wrap(sink, y + n);
 	}
 }
+
+/**
+ * \brief Volume passthrough from 32 bit to 32 bit.
+ * \param[in,out] dev Volume base component device.
+ * \param[in,out] sink Destination buffer.
+ * \param[in,out] source Input buffer.
+ * \param[in] frames Number of frames to process.
+ * \param[in] attenuation factor for peakmeter adjustment (unused)
+ *
+ * Copy and scale volume from 32 bit source buffer
+ * to 32 bit destination buffer.
+ */
+static void vol_passthrough_s32_to_s32(struct processing_module *mod,
+				       struct input_stream_buffer *bsource,
+				       struct output_stream_buffer *bsink, uint32_t frames,
+				       uint32_t attenuation)
+{
+	struct audio_stream __sparse_cache *source = bsource->data;
+	struct audio_stream __sparse_cache *sink = bsink->data;
+	int32_t *x;
+	int32_t *y;
+	int nmax, n;
+	const int nch = audio_stream_get_channels(source);
+	int remaining_samples = frames * nch;
+
+	x = audio_stream_wrap(source, (char *)audio_stream_get_rptr(source) + bsource->consumed);
+	y = audio_stream_wrap(sink, (char *)audio_stream_get_wptr(sink) + bsink->size);
+	bsource->consumed += VOL_S32_SAMPLES_TO_BYTES(remaining_samples);
+	bsink->size += VOL_S32_SAMPLES_TO_BYTES(remaining_samples);
+	while (remaining_samples) {
+		nmax = audio_stream_samples_without_wrap_s32(source, x);
+		n = MIN(remaining_samples, nmax);
+		nmax = audio_stream_samples_without_wrap_s32(sink, y);
```

Comment:
@andrula-song @singalsu 
Hmm,  audio_stream_samples_without_wrap_s24() is identical to audio_stream_samples_without_wrap_s32(), so effectively vol_passthrough_s32_to_s32() and vol_passthrough_s24_to_s24 are duplicated code. Can we just use one fucntion instead? 

And it seems generic copy seems like something we use in other audio functions, so do we have this function already somewhere else we could just call?